native/core/src/execution/planner.rs

```diff
@@ -818,6 +819,22 @@ impl PhysicalPlanner {
                 ));
                 Ok(array_join_expr)
             }
+            ExprStruct::ArrayExcept(expr) => {
```

Comment:
Because the PR only contains basic tests, could you add a check to enable this expression only if `CometConf.COMET_CAST_ALLOW_INCOMPATIBLE` is enabled? We can remove this check in a future PR that adds comprehensive tests and demonstrates that we have Spark-compatible behavior for all supported data types.

Reply:
COMET_CAST_ALLOW_INCOMPATIBLE check has been added.